extensions/benchmark/src/main/java/ai/djl/benchmark/MultithreadedBenchmark.java

```diff
@@ -41,16 +43,22 @@
 
         MemoryTrainingListener.collectMemoryInfo(metrics); // Measure memory before loading model
 
-        ZooModel<Void, float[]> model = loadModel(arguments, metrics);
-
-        int numOfThreads = arguments.getThreads();
+        Engine engine = Engine.getEngine(arguments.getEngine());
+        Device[] devices = engine.getDevices(arguments.getMaxGpus());
+        int numOfThreads = arguments.getThreads() * devices.length;
         int delay = arguments.getDelay();
         AtomicInteger counter = new AtomicInteger(iteration);
         logger.info("Multithreading inference with {} threads.", numOfThreads);
 
+        List<ZooModel<Void, float[]>> models = new ArrayList<>(devices.length);
         List<PredictorCallable> callables = new ArrayList<>(numOfThreads);
-        for (int i = 0; i < numOfThreads; ++i) {
-            callables.add(new PredictorCallable(model, metrics, counter, i, i == 0));
+        for (Device device : devices) {
```

Comment:
This means that the number of threads is really the number of threads per GPU. Is this the behavior users would expect? Should we document this somewhere or change some names to make it clearer?

Reply:
number of threads per GPU is confusing, I change it to total thread and added error checking,
I also changed number of iteration in the same way.